### ISSUE TITLE
Indeterminant linear system detected while working near variable

### ISSUE THREAD

**Issue by zhaoworking** [borglab/gtsam#964]
```
################ NEW FACTORS AT TIME 1626318282.750650 ################
NonlinearFactorGraph: size: 3

Factor 0: ImuFactor(x88,v88,x89,v89,b88)
preintegrated measurements:
    deltaTij 0.05
    deltaRij.ypr = (-4.58681042e-05 -0.000265717352  0.000372633753)
    deltaPij [-0.000490940523, 0.000735574774, 0.0120388693]'
    deltaVij [-0.0191920688, 0.0299153734, 0.482425743]'
    gyrobias [0, 0, 0]'
    acc_bias [0, 0, 0]'

    preintMeasCov 
[ 1.53125001e-09   1.2509485e-17  2.15427521e-18 -1.65205415e-16 -5.26200298e-12  3.35800697e-13 -1.04223302e-14 -3.33090037e-10  2.09902469e-11
  1.2509485e-17  1.53125002e-09 -1.53640589e-18  5.26191276e-12 -1.59239783e-16  2.07899034e-13  3.33084412e-10 -1.00285151e-14  1.28838896e-11
 2.15427521e-18 -1.53640589e-18  1.53125003e-09 -3.36785831e-13 -2.07192214e-13 -8.33493979e-17 -2.10526005e-11 -1.28391533e-11 -5.18923917e-15
-1.65205415e-16  5.26191276e-12 -3.36785831e-13   4.1562842e-09  8.47799052e-17  1.34069245e-15  1.25001812e-07  4.46190586e-15  7.02545197e-14
-5.26200298e-12 -1.59239783e-16 -2.07192214e-13  8.47799052e-17  4.15628412e-09  -2.1525147e-15  4.49788672e-15  1.25001807e-07 -1.14005568e-13
 3.35800697e-13  2.07899034e-13 -8.33493979e-17  1.34069245e-15  -2.1525147e-15  4.15625019e-09  7.11711822e-14 -1.14580265e-13   1.2500001e-07
-1.04223302e-14  3.33084412e-10 -2.10526005e-11  1.25001812e-07  4.49788672e-15  7.11711822e-14  5.00010227e-06  2.50441129e-13  3.96094251e-12
-3.33090037e-10 -1.00285151e-14 -1.28391533e-11  4.46190586e-15  1.25001807e-07 -1.14580265e-13  2.50441129e-13  5.00010201e-06  -6.4393506e-12
 2.09902469e-11  1.28838896e-11 -5.18923917e-15  7.02545197e-14 -1.14005568e-13   1.2500001e-07  3.96094251e-12  -6.4393506e-12  5.00000056e-06]
  noise model sigmas: 3.91311897e-05 3.91311898e-05 3.91311899e-05 6.44692501e-05 6.44692494e-05 6.44689863e-05  0.00223609084  0.00223609079   0.0022360681

Factor 1: BetweenFactor(b88,b89)
  measured: acc = [0, 0, 0]' gyro = [0, 0, 0]'
  noise model: diagonal sigmas[0.001; 0.001; 0.001; 1e-05; 1e-05; 1e-05];

Factor 2: BetweenFactor(x88,x89)
  measured: R:
[
	      0.9999987,  0.000816484903,  -0.00139033375;
 	-0.000815426224,     0.999999377,  0.000761853454;
 	  0.00139095493, -0.000760718749,     0.999998743
]
[-0.00238359758, -0.0981021114, 0.00579207297]';
  noise model: diagonal sigmas[2.15791864e-05; 2.15791864e-05; 4.37019372; 1; 1; 1];

time consuming per episode is 0.024776717
current_bias:  
acc = [9.8821951e+90, -1.17842217e+92, 1.21878767e+92]' gyro = [-7.14350686e+88, -2.34964682e+89, 2.48906269e+89]'
   0.999986527 -0.00517254398 0.000435631792   0.0236690054
  0.0051716756    0.999984683  0.00197145619   -0.267376937
-0.00044582256 -0.00196917668    0.999997962  0.00374918896
             0              0              0              1
    0.999986527  -0.00517254398  0.000435631791
   0.0051716756     0.999984683   0.00197145619
-0.000445822562  -0.00196917668     0.999997962
 0.0236690054
 -0.267376937
0.00374918896
cur_state:

|0.389296034, -0.3907936, -0.834103686|
|0.83344345, 0.535012545, 0.138324231|
|0.392199712, -0.749027329, 0.533982628|

[1.17262331e+105, -5.39972378e+104, 1.588715e+103]';

 7.16926278e+105
-7.30289123e+105
 1.31886773e+106
pro_state:

|-0.0999942018, 0.951112991, 0.292207526|
|-0.716640183, 0.134884699, -0.684275504|
|-0.690237646, -0.277831238, 0.6681181|

[1.53108645e+105, -9.05116939e+104, 6.75321015e+104]';

 7.16926278e+105
-7.30289123e+105
 1.31886773e+106
################ NEW FACTORS AT TIME 1626318282.800644 ################
NonlinearFactorGraph: size: 3

Factor 0: ImuFactor(x89,v89,x90,v90,b89)
preintegrated measurements:
    deltaTij 0.05
    deltaRij.ypr = (-0.000100834303 -0.000273347767  0.000322244532)
    deltaPij [-0.000406438029, 0.000778581261, 0.0120610778]'
    deltaVij [-0.0166465529, 0.0308472483, 0.483224267]'
    gyrobias [0, 0, 0]'
    acc_bias [0, 0, 0]'

    preintMeasCov 
[ 1.53125001e-09  1.10286108e-17  4.08221595e-18 -3.12288466e-16 -5.27593084e-12  3.39259137e-13 -1.96869388e-14   -3.335714e-10  2.10284621e-11
 1.10286108e-17  1.53125001e-09  -3.4139101e-18  5.27585033e-12 -2.96289762e-16  1.88993063e-13  3.33566402e-10 -1.87039765e-14  1.17694272e-11
 4.08221595e-18  -3.4139101e-18  1.53125002e-09 -3.40118676e-13 -1.88254832e-13  -7.6819317e-17 -2.10827976e-11 -1.17227737e-11 -4.77038733e-15
-3.12288466e-16  5.27585033e-12 -3.40118676e-13  4.15628446e-09  7.95541444e-17  1.23393154e-15  1.25001821e-07  4.14662021e-15  6.43741133e-14
-5.27593084e-12 -2.96289762e-16 -1.88254832e-13  7.95541444e-17  4.15628436e-09 -2.21258287e-15  4.13507925e-15  1.25001816e-07 -1.15191237e-13
 3.39259137e-13  1.88993063e-13  -7.6819317e-17  1.23393154e-15 -2.21258287e-15  4.15625019e-09  6.51218504e-14 -1.16858231e-13   1.2500001e-07
-1.96869388e-14  3.33566402e-10 -2.10827976e-11  1.25001821e-07  4.13507925e-15  6.51218504e-14  5.00010263e-06  2.29358628e-13  3.62000652e-12
  -3.335714e-10 -1.87039765e-14 -1.17227737e-11  4.14662021e-15  1.25001816e-07 -1.16858231e-13  2.29358628e-13  5.00010234e-06 -6.47612217e-12
 2.10284621e-11  1.17694272e-11 -4.77038733e-15  6.43741133e-14 -1.15191237e-13   1.2500001e-07  3.62000652e-12 -6.47612217e-12  5.00000054e-06]
  noise model sigmas: 3.91311897e-05 3.91311898e-05 3.91311899e-05  6.4469252e-05 6.44692513e-05 6.44689862e-05  0.00223609093  0.00223609086   0.0022360681

Factor 1: BetweenFactor(b89,b90)
  measured: acc = [0, 0, 0]' gyro = [0, 0, 0]'
  noise model: diagonal sigmas[0.001; 0.001; 0.001; 1e-05; 1e-05; 1e-05];

Factor 2: BetweenFactor(x89,x90)
  measured: R:
[
	    0.999986527,  -0.00517254398,  0.000435631791;
 	   0.0051716756,     0.999984683,   0.00197145619;
 	-0.000445822562,  -0.00196917668,     0.999997962
]
[0.0236690054, -0.267376937, 0.00374918896]';
  noise model: diagonal sigmas[2.15791864e-05; 2.15791864e-05; 4.37019372; 1; 1; 1];

time consuming per episode is 0.310579328
current_bias:  
acc = [1.23553509e+102, -3.31158791e+101, -4.0690101e+101]' gyro = [-3.58653312e+101, -3.06322487e+100, -1.30439877e+101]'
    0.999998129   0.00186719733   0.00050467284  0.000534461105
 -0.00186740992     0.999998168  0.000421095062    0.0510049192
-0.000503885653 -0.000422036711     0.999999784 -0.000119065414
              0               0               0               1
    0.999998129   0.00186719733  0.000504672842
 -0.00186740992     0.999998168  0.000421095065
 -0.00050388565 -0.000422036709     0.999999784
 0.000534461105
   0.0510049192
-0.000119065414
cur_state:

|-0.0722016239, 0.975940303, 0.205736364|
|-0.878035564, 0.0356530879, -0.477265551|
|-0.473117823, -0.215103192, 0.854336083|

[-inf, -nan, -nan]';

-1.64796567e+116
-1.32223026e+116
 1.63670331e+115
pro_state:

|-0.273264549, 0.246483681, 0.929823791|
|-0.946273615, 0.104855446, -0.305894721|
|-0.172895145, -0.963457903, 0.204587728|

[-inf, -nan, -nan]';

-1.64796567e+116
-1.32223026e+116
 1.63670331e+115
################ NEW FACTORS AT TIME 1626318282.850657 ################
NonlinearFactorGraph: size: 3

Factor 0: ImuFactor(x90,v90,x91,v91,b90)
preintegrated measurements:
    deltaTij 0.05
    deltaRij.ypr = (-4.27868961e-05 -0.000273358849  0.000160356884)
    deltaPij [-0.000426633634, 0.000757464668, 0.0121742035]'
    deltaVij [-0.0175318562, 0.0307383997, 0.486504015]'
    gyrobias [0, 0, 0]'
    acc_bias [0, 0, 0]'

    preintMeasCov 
[ 1.53125001e-09  5.28344283e-18  8.41751307e-19 -1.59573122e-16 -5.30114619e-12  3.38027147e-13 -1.01027617e-14 -3.34878024e-10  2.15171442e-11
 5.28344283e-18     1.53125e-09  -1.4188719e-18  5.30109411e-12 -1.27992729e-16  1.82869262e-13  3.34874601e-10 -8.15644165e-15   1.2438177e-11
 8.41751307e-19  -1.4188719e-18  1.53125001e-09 -3.38456072e-13 -1.82137468e-13 -6.08635408e-17 -2.15442587e-11 -1.23919454e-11 -3.93822457e-15
-1.59573122e-16  5.30109411e-12 -3.38456072e-13  4.15628478e-09  7.45523206e-17  1.17426949e-15  1.25001838e-07  4.20880175e-15  6.61679274e-14
-5.30114619e-12 -1.27992729e-16 -1.82137468e-13  7.45523206e-17  4.15628468e-09 -2.19825256e-15  4.00508824e-15  1.25001833e-07 -1.17203734e-13
 3.38027147e-13  1.82869262e-13 -6.08635408e-17  1.17426949e-15 -2.19825256e-15  4.15625018e-09  6.25320104e-14 -1.16388732e-13   1.2500001e-07
-1.01027617e-14  3.34874601e-10 -2.15442587e-11  1.25001838e-07  4.00508824e-15  6.25320104e-14  5.00010349e-06  2.41172803e-13  3.75979319e-12
-3.34878024e-10 -8.15644165e-15 -1.23919454e-11  4.20880175e-15  1.25001833e-07 -1.16388732e-13  2.41172803e-13   5.0001032e-06 -6.60964624e-12
 2.15171442e-11   1.2438177e-11 -3.93822457e-15  6.61679274e-14 -1.17203734e-13   1.2500001e-07  3.75979319e-12 -6.60964624e-12  5.00000056e-06]
  noise model sigmas: 3.91311897e-05 3.91311897e-05 3.91311898e-05 6.44692545e-05 6.44692537e-05 6.44689862e-05  0.00223609112  0.00223609105   0.0022360681

Factor 1: BetweenFactor(b90,b91)
  measured: acc = [0, 0, 0]' gyro = [0, 0, 0]'
  noise model: diagonal sigmas[0.001; 0.001; 0.001; 1e-05; 1e-05; 1e-05];

Factor 2: BetweenFactor(x90,x91)
  measured: R:
[
	    0.999998129,   0.00186719733,  0.000504672842;
 	 -0.00186740992,     0.999998168,  0.000421095065;
 	 -0.00050388565, -0.000422036709,     0.999999784
]
[0.000534461105, 0.0510049192, -0.000119065414]';
  noise model: diagonal sigmas[2.15791864e-05; 2.15791864e-05; 4.37019372; 1; 1; 1];

terminate called after throwing an instance of 'gtsam::IndeterminantLinearSystemException'
  what():  
Indeterminant linear system detected while working near variable
7061644215716937818 (Symbol: b90).

Thrown when a linear system is ill-posed.  The most common cause for this
error is having underconstrained variables.  Mathematically, the system is
underdetermined.  See the GTSAM Doxygen documentation at
http://borg.cc.gatech.edu/ on gtsam::IndeterminantLinearSystemException for
more information.
Aborted
pid: 
../script/run.sh: 1: kill: Usage: kill [-s sigspec | -signum | -sigspec] [pid | job]... or
kill -l [exitstatus]

```

Here I use IMU between Factor and 6DOF visual odometry pose Between factor  to construct my graph using isam2 to update it. But I found this problem after a short time. Obviously, the IMU bias displayed which after optimization is too big. But I don't know what cause the bias increases in such a pace that reaches at **1e91**  magnitude that cause the system just shut down. 
So,What is the root reason of such problem? And for sake of the convenience and intuition , I past the code blow.
```
            new_factors.emplace_shared<BetweenFactor<Pose3>>(
                previous_pose_key,
                current_pose_key,
                Pose3(Rot3(imu_cur_T_pre_.rotation()), imu_cur_T_pre_.translation()),
                noise_model_pose);
```
```

            new_factors.emplace_shared<ImuFactor>(previous_pose_key, previous_vel_key,
                                                  current_pose_key, current_vel_key,
                                                  previous_bias_key, *current_summarized_measurement);
```
```
            new_factors.emplace_shared<BetweenFactor<imuBias::ConstantBias>>(previous_bias_key,
                                                                             current_bias_key,
                                                                             imuBias::ConstantBias(),
                                                                             sigma_between_b);
```

**Comment by dellaert** [comment]
Not a bug, please subscribe to Google group and ask there. But even there I think no one will be able to help you based on only this information. Working with IMU factors is hard and people might only help you if they know the steps you have taken to debug etc.

**Comment by zhaoworking** [comment]
> 

Thanks a lot whatsoever. Notable to say, I change the 6DOF factor pose_noise of which the orientation noise is set to INT_MAX ,to the zero. And , it works fine. But I am confused with this noise model, could u tell me what the role that the noise parameter here represent in the noise model. Cause when I set INT_MAX, i found the noise model is [0,0,0], however , when I set zeros, i found it is [inf ,inf, inf] in `factor.print()`
I just wanna know what is the matter with this noise parameter. :-)
